tests/test_transformations.py

```diff
@@ -127,6 +160,218 @@ class AllStrict(Mapping):
             expected = {"obj": {"target": 1}}
             self.assertEqual(expected, result)
 
+    def test_all_transformation_strict(self):
+        class User(Schema):
+            name = UnicodeField()
+
+        class Message(Schema):
+            id = IntegerField()
+            user = Subschema(User)
+
+        class Transformed(Schema):
+            user_name = UnicodeField()
+            object = ObjectField()
+
+        class SomeNonStrictMapping(Mapping):
+            source_schema = Message
+            target_schema = Transformed
+
+            user_name = Get('user', 'name')
+            object = All(strict=False)
+
+        class SomeStrictMapping(Mapping):
+            source_schema = Message
+            target_schema = Transformed
+
+            user_name = Get('user', 'name')
+            object = All(strict=True)
+
+        non_strict = SomeNonStrictMapping().apply(self.basic_data).serialize()
+        strict = SomeStrictMapping().apply(self.basic_data).serialize()
+
+        self.assertDictEqual(self.non_strict_output, non_strict)
+        self.assertDictEqual(self.strict_output, strict)
+
+    def test_all_transformation_generic_schema(self):
+
+        class SomeNonStrictMapping(Mapping):
+            user_name = Get('user', 'name')
+            object = All(strict=False)
+
+        class SomeStrictMapping(Mapping):
+            user_name = Get('user', 'name')
+            object = All(strict=True)
+
+        non_strict = SomeNonStrictMapping().apply(self.basic_data).serialize()
+        strict = SomeStrictMapping().apply(self.basic_data).serialize()
+
+        # All fields should be included, regardless of the strict param
+        self.assertDictEqual(self.non_strict_output, non_strict)
+        self.assertDictEqual(self.non_strict_output, strict)
+
+    def test_all_on_nested_generic_schema(self):
+        nested_data = {
+            'user': {
+                'name': 'Peanut',
+                'profile_img': 'lala.jpg',
+                'extra': {
+                    'slug': 'Snoopy is amazing.',
+                    'not_defined': 'should still show up',
+                    'sub_extra': {
+                        'deep': 2,
+                        'x': 'y'
+                    }
+                }
+            }
+        }
+        nested_strict_data = {
+            'user': {
+                'name': 'Peanut',
+                'extra': {
+                    'slug': 'Snoopy is amazing.',
+                    'sub_extra': {
+                        'deep': 2,
+                    }
+                }
+            }
+        }
+
+        class SubExtra(Schema):
+            deep = IntegerField()
+
+        class SourceExtra(Schema):
+            slug = UnicodeField()
+            sub_extra = Subschema(SubExtra)
+
+        class SourceUser(Schema):
+            name = UnicodeField()
+
+            # All() should know how to deal with a Schema within
+            # a GenericSchema
```

Comment:
Ditto above; if this is all you're trying to test, then much of this 65 line test is just noise.